app/backend/approaches/chatreadretrieveread.py

```diff
@@ -242,3 +264,17 @@ def get_messages_from_history(
 
         messages = message_builder.messages
         return messages
+    
+    def get_search_query(self, chat_completion: any, user_query: str):
+        response_message = chat_completion["choices"][0]["message"]
+        print(response_message)
+        if function_call := response_message.get("function_call"):
+            if function_call["name"] == "search_sources":
+                arg = json.loads(function_call["arguments"])
+                search_query = arg["search_query"]
```

Comment:
Will search_query always be defined? I see you check is not None in the next if, but you use square brackets here instead of get(), so that implies it's always defined but could be None. Is that the case?